specs/requests/artifacts_spec.py

```diff
@@ -0,0 +1,278 @@
+""" Tests for artifacts """
+
+import sys
+from io import BytesIO
+from flask import current_app
+from mamba import description, context, before, after, it
+from expects import expect, equal, have_key
+from elasticsearch.exceptions import NotFoundError
+from doublex import Mock, Stub, ANY_ARG
+from specs.spec_helpers import Context
+
+sys.path.insert(0, 'specs')
+
+with description('/images') as self:
+
+    # pylint: disable=duplicate-code
+    with before.each:
+        self.context = Context()
+        current_app.es = Stub()
+
+    with after.each:
+        self.context.delete()
+    # pylint: enable=duplicate-code
+
+    with description('/'):
+        with description('GET without database'):
+            with before.each:
+                current_app.es = None
+                self.response = self.context.client().get("/images")
+
+            with it('returns a 503 status code'):
+                expect(self.response.status_code).to(equal(503))
+
+        with description('GET'):
+            with before.each:
+                with Mock() as elastic_mock:
+                    elastic_mock.search(ANY_ARG).returns(
+                        {"hits": {"total": 12, "max_score": 1.0, "hits": [
+                            {"_index": "artifact",
+                             "_type": "image",
+                             "_id": 1,
+                             "_score": 1.0,
+                             "_source": {
+                                 "created_at": "today",
+                                 "updated_at": "today",
+                                 "file_url": "class_diagram.png",
+                                 "tags": ["uml", "class diagram"],
+                                 "file_date": "today"}},
+                            {"_index": "artifact",
+                             "_type": "image",
+                             "_id": 1,
+                             "_score": 0.5,
+                             "_source": {
+                                 "created_at": "yesterday",
+                                 "updated_at": "yesterday",
+                                 "file_url": "use_case_diagram.png",
+                                 "tags": ["uml", "use case diagram"],
+                                 "file_date": "yesterday"}}]}})
+
+                current_app.es = elastic_mock
+                self.response = self.context.client().get("/images")
+
+            with it('returns a 200 status code'):
+                expect(self.response.status_code).to(equal(200))
```

Comment:
This doesn'T check for the response format and also doesn't check if offset, limit and search term have been passed correctly.